quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_type.rs

```diff
@@ -71,6 +71,38 @@ impl Default for QuickwitDateTimeOptions {
 }
 
 impl QuickwitDateTimeOptions {
+    pub(crate) fn validate_json(
+        &self,
+        json_value: &serde_json_borrow::Value,
+    ) -> Result<(), String> {
+        match json_value {
+            serde_json_borrow::Value::Number(timestamp) => {
+                // `.as_f64()` actually converts floats to integers, so we must check for integers
+                // first.
+                if let Some(timestamp_i64) = timestamp.as_i64() {
+                    quickwit_datetime::parse_timestamp_int(timestamp_i64, &self.input_formats.0)?;
+                    Ok(())
+                } else if let Some(timestamp_f64) = timestamp.as_f64() {
+                    quickwit_datetime::parse_timestamp_float(timestamp_f64, &self.input_formats.0)?;
+                    Ok(())
+                } else {
+                    Err(format!(
+                        "failed to parse datetime `{:?}`: value is larger than i64::MAX",
```

Comment:
this code is unreachable since `as_f64` always succeeds

Reply:
I can't unwrap though, always returning `Some` is an implementation detail of serde_borrow_json. I'll change the error message.